### DISCUSSION TITLE
State wallet and backup offering

### DISCUSSION THREAD

**Discussion by tobiaschristen** [e-id-admin/open-source-community#24] [General]
When I read about the wallet that Switzerland wants to launch and that it will feature a backup solution as well, I was wondering what the scope will be: 

-	Does it support ambition level 3 verifiable credentials the same way it supports state credentials? I.e. is the ambition of the wallet that it is the only wallet that the average citizen will need?

-	Does the “backup” intention mean that the state wants to offer a “hybrid wallet” where data is stored encrypted in the state cloud and only keys are on the mobile device?

-	Will the state backup cloud be open to other wallets as well?

-	Does the state wallet intend to offer offline access as well as online access to VCs ?

-	Will the state wallet be open source?

**Comment by christianheimann** [comment]
1. There will be a wallet which is provided by the Confederation. With this wallet, users will be able to receive, store and present all kind of credentials (ambition level 3). Additionally the wallet will contain the necessary logic for the online process to get the E-ID: this said, a user who wants to do the online process will need the wallet app from the Confederation, but the target wallet for the then issued E-ID can be another one. 
2. Current discussion and concepts do not point to a "hybrid wallet". The planned backup is more to be seen as a datastorage, to which the wallet can send an encrypted data package and later another instance of the wallet can download it and reuse the data.
3. The backup service provided by the Confederation is a "feature of the wallet", so there will be no other wallets that can use this service.
4. As the wallet is not hybrid and credentials will be stored on the device itself, the acces to VCs exists at any time. The general offline usage expectations are actually discussed and also depend on the underlaying tech-stack.
5. As proposed in the [Vernehmlassung](https://www.bj.admin.ch/bj/de/home/aktuell/mm.msg-id-89515.html), the wallet provided by the Confederation shall be open source.

I hope these answers give you an idea. Any concrete inputs, thoughts or potential pitfalls will be warmly welcomed!

**Reply by pascalgottret** [reply]
Hi Christian, it seems that the link for the Vernehmlassung does not work anymore. Do you have another link, where there is a written comment on the open-source wallet?

**Reply by christianheimann** [reply]
Thx, Link is fixed.
As the message ([Botschaft](https://www.fedlex.admin.ch/eli/fga/2023/2842/de)) was adopted by the federal council end 2023, there are further comments on the concerning open source articles. See also the discussion #27 > [Comments on open source topics](https://github.com/e-id-admin/open-source-community/discussions/27#discussioncomment-7791096)
